### PR TITLE
LogData-Dillon

### PR DESCRIPTION
**Pull request by NataliaPavlovic**

### CODE REVIEW

File: Src/LogData.c [hunk 1]
```diff
@@ -216,13 +218,35 @@ void logDataTask(void const* arg)
 
     if (f_mount(&fatfs, "SD:", 1) == FR_OK)
     {
-        HAL_GPIO_WritePin(LED1_GPIO_Port, LED1_Pin, 1);
-
-        if (f_open(&file, "SD:VanderAvionics.csv", FA_OPEN_APPEND | FA_READ | FA_WRITE) == FR_OK)
+        if (f_open(&file, "SD:VanderAvionics.csv", FA_OPEN_EXISTING) == FR_NO_FILE)
         {
+            f_open(&file, "SD:VanderAvionics.csv", FA_CREATE_NEW | FA_READ | FA_WRITE);
```
Comment (NataliaP721) [inline Src/LogData.c]: Add line HAL_GPIO_WritePin(LED1_GPIO_Port, LED1_Pin, 1) here. We want LED1 to turn on when we create a file.

File: Src/LogData.c [hunk 2]
```diff
@@ -216,13 +218,35 @@ void logDataTask(void const* arg)
...
             f_puts(buffer, &file);
             f_close(&file);
         }
+        else
+        {
+            uint8_t exists = 1;
```
Comment (NataliaPavlovic) [inline Src/LogData.c]: I would rename this variable to fileExists so it is more clear what it is for.

File: Src/LogData.c [hunk 3]
```diff
@@ -216,13 +218,35 @@ void logDataTask(void const* arg)
 
     if (f_mount(&fatfs, "SD:", 1) == FR_OK)
     {
-        HAL_GPIO_WritePin(LED1_GPIO_Port, LED1_Pin, 1);
-
-        if (f_open(&file, "SD:VanderAvionics.csv", FA_OPEN_APPEND | FA_READ | FA_WRITE) == FR_OK)
+        if (f_open(&file, "SD:VanderAvionics.csv", FA_OPEN_EXISTING) == FR_NO_FILE)
         {
```
Comment (NataliaPavlovic) [inline Src/LogData.c:222]: Add line HAL_GPIO_WritePin(LED1_GPIO_Port, LED1_Pin, 1) here. We want LED1 to turn on when we create a file.

**Comment by NataliaPavlovic** [comment]
I opened this pull request, so I cannot add myself as reviewer. I approve of these changes. I tested these changes and have attached the logs.
[Logs.zip](https://github.com/StudentOrganisationForAerospaceResearch/AvionicsSoftware/files/2964503/Logs.zip)

**Pull request merged**